src/yunohost/app.py

```diff
@@ -669,55 +666,81 @@ def app_upgrade(app=[], url=None, file=None):
 
         # Execute App upgrade script
         os.system('chown -hR admin: %s' % INSTALL_TMP)
-        if hook_exec(extracted_app_folder + '/scripts/upgrade',
-                     args=args_list, env=env_dict)[0] != 0:
-            msg = m18n.n('app_upgrade_failed', app=app_instance_name)
-            operation_logger.error(msg)
-
-            # display this if there are remaining apps
-            if apps[number + 1:]:
-                logger.error(m18n.n('app_upgrade_stopped'))
-                not_upgraded_apps = apps[number:]
-                # we don't want to continue upgrading apps here in case that breaks
-                # everything
-                raise YunohostError('app_not_upgraded',
-                                    failed_app=app_instance_name,
-                                    apps=', '.join(not_upgraded_apps))
+
+
+        try:
+            upgrade_retcode = hook_exec(extracted_app_folder + '/scripts/upgrade',
+                                        args=args_list, env=env_dict)[0]
+        except (KeyboardInterrupt, EOFError):
+            upgrade_retcode = -1
+        except Exception:
+            import traceback
+            logger.exception(m18n.n('unexpected_error', error=u"\n" + traceback.format_exc()))
+        finally:
+
+            # Did the script succeed ?
```

Comment:
Shouldn't we had a test for `-1` saying "the user stopped the installation"?

Reply:
875c570 should fix it :+1:

Reply:
Thx!